File: grind75/Dec20/53.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        largest = -math.inf
        cumulative_sum = 0

        for num in nums:
            if cumulative_sum <= 0:
                cumulative_sum = num
            else:
                cumulative_sum += num
            
            largest = max(largest, cumulative_sum)
        
        return largest

File: grind75/Dec20/53.go
func maxSubArray(nums []int) int {
    ret := math.MinInt
    cur := 0
    for _, val := range nums {
        if cur <= 0 { 
            cur = val 
        } else { 
            cur = val+cur 
        }
        ret = max(ret, cur)
    }
    return ret
}
